astropy/io/fits/hdu/hdulist.py

```diff
@@ -930,27 +936,30 @@ def info(self, output=None):
         else:
             name = self._file.name
 
-        results = ['Filename: {}'.format(name),
-                   'No.    Name      Ver    Type      Cards   Dimensions   Format']
-
-        format = '{:3d}  {:10}  {:3} {:11}  {:5d}   {}   {}   {}'
+        file_line = 'Filename: {}\n'.format(name)
+        results = []
         default = ('', '', '', 0, (), '', '')
         for idx, hdu in enumerate(self):
-            summary = hdu._summary()
-            if len(summary) < len(default):
-                summary += default[len(summary):]
-            summary = (idx,) + summary
+            hdu_summary = hdu._summary()
+            summary = [idx, *hdu_summary, *default[len(hdu_summary):]]
             if output:
-                results.append(format.format(*summary))
+                results.append(tuple(map(str, summary)))
             else:
-                results.append(summary)
+                results.append(tuple(summary))
 
         if output:
-            output.write('\n'.join(results))
+            t = Table(rows=results, names=COLUMN_NAMES)
+            t_strings = t.pformat(max_lines=-1, max_width=-1, show_dtype=False,
+                                  align=COLUMN_ALIGN)
```

Comment:
Ok for `max_lines`, but I would prefer `max_width=None` to have the output fit the screen width (hiding some columns). Maybe we could also allow to pass additional kwargs ? To force the `max_width` value, or use HTML output in the notebook (not sure of the result with the following lines though).

Reply:
I was thinking about backwards-compatibility here. Without that it will, as you said, possibly truncate the output. Especially "wide-tables" are affected here.

Reply:
Right, but I had #821 in mind: having long lines which wraps is not readable. 
Also I don't know if there is a way to tell `Table` which column should be hidden first (I guess it's not possible currently).

Reply:
Hm, there might be other alternatives. Table-Columns allow for a `format` attribute. One could truncate the contents or wrap the lines. Both not really hard. But personally I would almost prefer an option to simply return the Table instead of printing it. Unfortunately the `output=False` option is already taken and I don't know if I want to break it by returning the Table.

Reply:
@saimn I like the idea of fitting to the screen width, but that could potentially be difficult to do in a portable way. Maybe `curses` could help?

@MSeifert04 what about adding an option called `display` that is `True` by default but can be set to `False` to return the table?

Reply:
Windows doesn't have curses...

Reply:
^ This looks like an opening for a joke about curses and Windows, but I'll refrain...

Reply:
@drdavella Yeah, that would be one possibility, I was also thinking about letting the `output` argument be a string like `Table` which indicates "return a `Table`".

Reply:
I think there are ways to [make curses work on windows](https://stackoverflow.com/a/41224335), but I guess that gets back to it being "difficult to in a portable way."

The idea of `output='table'` seems pretty reasonable to me.